tests/loadgen/data.go

```diff
@@ -181,6 +182,43 @@ func VerifyData(data DataGen) {
     }
 }
 
+func VerifyUserPushPermissions(baseURL, portNumber string, data DataGen) {
+    sessionID := initialLogin()
+    users := ListUsers(baseURL, sessionID)
+    projects := ListProjects(baseURL, sessionID)
+
+    rand.Seed(time.Now().UTC().UnixNano())
+    projectIndex := rand.Intn(len(projects))
+    userIndex := rand.Intn(len(users))
+    for projects[projectIndex].OwnerID == users[userIndex].UserID {
+        userIndex = rand.Intn(len(users))
+    }
+    user := users[userIndex]
+    imgVersion := data.GetImageVersion()
+
+    url := parseHarborURL(baseURL)
+    defaultPwd := "Harbor12345"
+    fmt.Println("Trying to push image without adequate permissions")
+    imgPushed := pushImageToRepo(user.Username, defaultPwd, url, portNumber, projects[projectIndex].Name, imgVersion.ImageName, imgVersion.TagName)
+    if !imgPushed {
+        fmt.Printf("Image push failed.\nApplying adequate permissions for the user\n")
+        memberReq := MemberReq {
+            Username: user.Username,
+            Roles   : []int{ 2 }, // developer role
+        }
+        // TODO: Check response code for permissions already set (409)
```

Comment:
Can you explain this todo a little bit? How does this happen if we failed to push the image to get to here?